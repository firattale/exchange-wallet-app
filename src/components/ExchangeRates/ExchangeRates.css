.exchange-container {
    color: white;
    background-color: #3762c7;
    border: 1px solid #3c70e0;
    padding: 7px;
    border-radius: 7px;
}

.currency {
    font-size: 13px;
}