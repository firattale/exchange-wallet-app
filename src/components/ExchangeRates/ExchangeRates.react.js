import React, { useEffect } from 'react'
import './ExchangeRates.css';
import { useSelector, useDispatch } from 'react-redux';
import {
    selectFirstCurrency,
    selectSecondCurrency,
    selectCurrencyRate,
    selectFirstSign,
    selectSecondSign,
    changeCurrencyRateAsync
} from '../../app/exchangeSlice';

export const ExchangeRates = () => {
    const first = useSelector(selectFirstCurrency);
    const second = useSelector(selectSecondCurrency);
    const currencyRate = useSelector(selectCurrencyRate);
    const firstSign = useSelector(selectFirstSign);
    const secondSign = useSelector(selectSecondSign);
    const dispatch = useDispatch();

    useEffect(() => {
        dispatch(changeCurrencyRateAsync({ first, second }));
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [first, second])
    return (
        <div className="exchange-container">
            <span className="currency">{firstSign}</span>
            1=
            <span className="currency">{secondSign}</span>
            {currencyRate.slice(0, 4)}
            <span className="currency">{currencyRate.slice(4, 6)}</span>
        </div>)
}

export default ExchangeRates;