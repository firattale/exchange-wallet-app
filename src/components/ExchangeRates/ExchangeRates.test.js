import React from 'react';
import { shallow } from 'enzyme';
import { ExchangeRates } from './ExchangeRates.react';
import { useSelector, useDispatch } from 'react-redux';
import {
    selectFirstCurrency,
    selectSecondCurrency,
    selectCurrencyRate,
    selectFirstSign,
    selectSecondSign,
    changeCurrencyRateAsync
} from '../../app/exchangeSlice';

jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));
describe('ExchangeRates', () => {
    const mockAppState = {
        exchange: {
            firstCurrency: "GBP",
            firstSign: "Â£",
            secondCurrency: "USD",
            currencyRate: "",
            secondSign: "$"
        },
        wallet: {
            USD: 40.00,
            GBP: 50.00,
            EUR: 60.00,
            firstAmount: "",
            secondAmount: "",
            firstWalletError: null
        }
    }
    beforeEach(() => {
        useSelector.mockImplementation(callback => {
            return callback(mockAppState);
        });
    });
    afterEach(() => {
        useSelector.mockClear();
    });
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<ExchangeRates debug />);

        expect(component).toMatchSnapshot();
    });

    it('should call useSelector 5 times', () => {
        shallow(<ExchangeRates debug />);
        expect(useSelector).toHaveBeenCalledTimes(5)
        expect(useDispatch).toHaveBeenCalledTimes(2)
        expect(useSelector).toHaveBeenCalledWith(selectFirstCurrency)
        expect(useSelector).toHaveBeenCalledWith(selectSecondCurrency)
        expect(useSelector).toHaveBeenCalledWith(selectFirstSign)
        expect(useSelector).toHaveBeenCalledWith(selectSecondSign)
        expect(useSelector).toHaveBeenCalledWith(selectCurrencyRate)
    });
 
});