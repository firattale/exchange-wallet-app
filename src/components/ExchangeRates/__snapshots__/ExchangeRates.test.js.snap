// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeRates should render correctly in "debug" mode 1`] = `
<div
  className="exchange-container"
>
  <span
    className="currency"
  >
    Â£
  </span>
  1=
  <span
    className="currency"
  >
    $
  </span>
  <span
    className="currency"
  />
</div>
`;
